hazelcast/src/test/java/com/hazelcast/nio/serialization/compatibility/APortable.java

```diff
@@ -358,6 +406,21 @@ public boolean equals(Object o) {
         if (str != null ? !str.equals(that.str) : that.str != null) {
             return false;
         }
+        if (!Objects.equals(bd, that.bd)) {
+            return false;
+        }
+        if (ld != that.ld) {
```

Comment:
I think you need to use `Objects.equals` for these fields as well (ld, lt, ldt, odt)

Reply:
Right, we need to use `Objects.equals` o/w test are failing, fixed at https://github.com/hazelcast/hazelcast/pull/22716/commits/53d396ba98264c8202a4bb8090d087cf3fcb67fc